internal/server/productstatus.go

```diff
@@ -0,0 +1,72 @@
+package server
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/metno/go-mms/pkg/mms"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type Product struct {
+	Name                 string
+	NextInstanceExpected time.Time
+}
+
+type Productstatus struct {
+	Products map[string]Product
+	gaugeVec *prometheus.GaugeVec
+}
+
+func NewProductstatus(m *metrics) *Productstatus {
```

Comment:
Should be NewProductStatus not NewProductstatus

Reply:
It is a _Productstatus_ component here, which keeps track of multiple products, not a product status.